api/lib/domain/usecases/get-campaign-participation-result.js

```diff
@@ -16,14 +18,14 @@ module.exports = async function getCampaignParticipationResult(
   await _checkIfUserHasAccessToThisCampaignParticipation(userId, campaignParticipation, campaignRepository);
 
   const targetProfile = await targetProfileRepository.getByCampaignId(campaignParticipation.campaignId);
-  const badge = await badgeRepository.findOneByTargetProfileId(targetProfile.id);
+  const campaignBadges = await badgeRepository.findByTargetProfileId(targetProfile.id);
+  const emploiCleaBadge = _.find(campaignBadges, (badge) => !_.isEmpty(badge.badgePartnerCompetences) && badge.key === Badge.keys.PIX_EMPLOI_CLEA);
 
-  const campaignParticipationResult = await campaignParticipationResultRepository.getByParticipationId(campaignParticipationId, badge);
+  const hasAcquiredBadgesList = await Promise.all(campaignBadges.map((badge) => badgeAcquisitionRepository.hasAcquiredBadgeWithId({ userId, badgeId: badge.id })));
+  const acquiredBadges = campaignBadges.filter((badge, index) => hasAcquiredBadgesList[index]);
```

Comment:
Un `Promise.all` a le potentiel d'entraîner une utilisation excessive de ressources au détriment des autres requêtes. Ici on lancerait autant de requêtes SQL simultanées qu'il y a de badges ; a priori ce nombre n'est pas très grand mais pour le principe, mieux vaut faire autrement.

On peut utiliser une fonctions utilitaire de `bluebird`, par exemple :

```suggestion
  const acquiredBadges = await bluebird.filter(campaignBadges, (badge) => 
    badgeAcquisitionRepository.hasAcquiredBadgeWithId({ userId, badgeId: badge.id }),
    { concurrency: 1 });
```

Ou passer à un `for` :

```suggestion
  const acquiredBadges = [];
  for (const badge of campaignBadges) {
  	if (await badgeAcquisitionRepository.hasAcquiredBadgeWithId({ userId, badgeId: badge.id })) {
  	  acquiredBadges.push(badge);
  	}
  }
```

Mais de toute façon, lancer une requête SQL par badge ce n'est vraiment pas efficace. Il faudrait plutôt que le `badgeAcquisitionRepository` permette d'interroger l'acquisition de plusieurs badges en une fois (ex. `hasAcquiredBadgesWithIds({ userId, badgeIds: [ … ]})`) ou éventuellement qu'il permette de récupérer tous les badges d'un utilisateur et faire le filtre côté JS.

Reply:
J'ai ajouté la fonction `badgeAcquisitionRepository.getAcquiredBadgeIds({ badgeIds, userId })` qui retourne les ids des bagdes obtenus. Ca permet d'enlever le `Promise.all`.